Is this not related to your changes? 

```

#12 1.496 --- stderr: nav2_common
#12 1.496 CMake Error at CMakeLists.txt:7 (find_package):
#12 1.496   By not providing "Findbackward_ros.cmake" in CMAKE_MODULE_PATH this project
#12 1.496   has asked CMake to find a package configuration file provided by
#12 1.496   "backward_ros", but CMake did not find one.
#12 1.496 
#12 1.496   Could not find a package configuration file provided by "backward_ros" with
#12 1.496   any of the following names:
#12 1.496 
#12 1.496     backward_rosConfig.cmake
#12 1.496     backward_ros-config.cmake
#12 1.496 
#12 1.496   Add the installation prefix of "backward_ros" to CMAKE_PREFIX_PATH or set
#12 1.496   "backward_ros_DIR" to a directory containing one of the above files.  If
#12 1.496   "backward_ros" provides a separate development package or SDK, be sure it
#12 1.496   has been installed.
#12 1.496 
#12 1.496 
#12 1.496 ---
#12 1.497 Failed   <<< nav2_common [0.53s, exited with code 1]

```

Its a dep that isn't being found... seems related (?)

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -56,6 +56,7 @@ RUN if [ ! -f /etc/ros/rosdep/sources.list.d/20-default.list ]; then \
 
 RUN apt update && apt upgrade -y \
     && rosdep install -y --ignore-src --from-paths src -r \
+       -t build -t buildtool -t build_export -t buildtool_export -t exec \
        --skip-keys "slam_toolbox turtlebot3_gazebo" \
     && rm -rf /var/lib/apt/lists/*
 
@@ -64,7 +65,7 @@ ARG BUILD=true
 ARG COLCON_BUILD_ARGS=""
 RUN if [ "${BUILD}" = "true" ]; then \
       . /opt/ros/${ROS_DISTRO}/setup.sh \
-      && colcon build $COLCON_BUILD_ARGS; \
+      && colcon build --cmake-args -DBUILD_TESTING=OFF $COLCON_BUILD_ARGS; \
     else \
       mkdir -p /root/nav2_ws/install; \
     fi
@@ -103,6 +104,7 @@ RUN if [ ! -f /etc/ros/rosdep/sources.list.d/20-default.list ]; then \
 
 RUN apt update && apt upgrade -y \
     && rosdep install -y --ignore-src --from-paths src -r \
+       -t build -t buildtool -t build_export -t buildtool_export -t exec \
        --skip-keys "slam_toolbox \
                     turtlebot3_gazebo \
                     rviz2 \
@@ -122,6 +124,7 @@ ARG COLCON_BUILD_ARGS=""
 RUN if [ "${BUILD}" = "true" ]; then \
       . /opt/ros/${ROS_DISTRO}/setup.sh \
       && colcon build \
+     --cmake-args -DBUILD_TESTING=OFF \
          --packages-skip nav2_rviz_plugins \
                          nav2_bringup \
                          nav2_system_tests \
@@ -199,7 +202,7 @@ RUN apt update && \
       rosdep init; \
     fi && rosdep update && \
     rosdep install -y --ignore-src --from-paths src -r --rosdistro ${ROS_DISTRO} \
-    --dependency-types=exec \
+    -t exec \
     --skip-keys "rviz2 \
                  gazebo_ros_pkgs \
                  ros_gz \

--- a/docker/Dockerfile.arm64
+++ b/docker/Dockerfile.arm64
@@ -48,6 +48,7 @@ RUN if [ ! -f /etc/ros/rosdep/sources.list.d/20-default.list ]; then \
 
 RUN apt-get update \
     && rosdep install -y --ignore-src --from-paths src -r \
+       -t build -t buildtool -t build_export -t buildtool_export -t exec \
        --skip-keys "slam_toolbox \
                     turtlebot3_gazebo \
                     gazebo_ros_pkgs \
@@ -69,6 +70,7 @@ ARG COLCON_BUILD_ARGS=""
 RUN if [ "${BUILD}" = "true" ]; then \
       . /opt/ros/${ROS_DISTRO}/setup.sh \
       && colcon build \
+         --cmake-args -DBUILD_TESTING=OFF \
          --packages-skip nav2_rviz_plugins \
                          nav2_bringup \
                          nav2_system_tests \

